pushdeploy/apiv1.py

```diff
@@ -25,26 +12,71 @@ def init_api():
     """Creates instances of the incluster config and client API
     and stores them in global"""
     g.configuration = config.load_incluster_config()
-    g.api_instance = client.AppsV1Api(client.ApiClient(g.configuration))
+    g.apps_v1_api_instance = client.AppsV1Api(client.ApiClient(g.configuration))
+    g.batch_v1beta1_instance = client.BatchV1beta1Api(client.ApiClient(g.configuration))
     g.PD_REGISTRY = current_app.config['PD_REGISTRY']
 
-def read_deployment(deployment, namespace):
+def list_cron_job(name, namespace):
+    namespace = "%s" % str(namespace)
+    name = "metadata.name=%s" % str(name)
+    api_response = g.batch_v1beta1_instance.list_namespaced_cron_job(
+        namespace=namespace,
+        field_selector=name
+    )
+    if len(api_response.items) == 1:
+        return api_response.items[0]
+    else:
```

Comment:
I'm not sure if returning a string for the `else` clause is the best approach. Your consumer of that calls a long chain of methods, e.g. `cron_job_object.spec.job_template.spec.template.spec.containers[0].image` which will probably blow up if `cron_job_object` were a string.

Do you want it to raise an exception as opposed to returning a string? I'm not sure if that's considered best practice in Python, but that might be an option.